pkg/codec/utils.go

```diff
@@ -13,6 +13,58 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/types"
 )
 
+var (
+	codecNumType = reflect.TypeOf(Number("0"))
+	intType      = reflect.TypeOf(int(0))
+	int8Type     = reflect.TypeOf(int8(0))
+	int16Type    = reflect.TypeOf(int16(0))
+	int32Type    = reflect.TypeOf(int32(0))
+	int64Type    = reflect.TypeOf(int64(0))
+	float32Type  = reflect.TypeOf(float32(0))
+	float64Type  = reflect.TypeOf(float64(0))
+)
+
+func NumberHook(from reflect.Type, to reflect.Type, val any) (any, error) {
+	if from == codecNumType && to != codecNumType {
```

Comment:
do we need to go the other way too?  Also, what is using this (or will it get used in chainlink and chainlink-solana for now?)

Reply:
The `chainlink` repo is using this for now and was only needed on the non-loop tests.

Reply:
Won't you need to convert back to a Number later?  This only goes from Number => represented type, but doesn't go the other way.